yetanotherco/aligned_layer
# fix: solidity coding style

# Description

- Adds `solhint` linter to the CI. To run `solhint` locally, go to `contracts`, run `npm install` and then `npm run lint:sol`. The CI will run the linter but won't stop the user from pushing to main, it just lists the warnings.
- Fixes most linter warnings, the ones left don't seem relevant.
- The three rules added were taken from [https://github.com/lambdaclass/era-contracts/blob/main/.solhint.json](https://github.com/lambdaclass/era-contracts/blob/main/.solhint.json) and it made sense to add them to this repo too.
- Add `make lint_contracts` to makefile

# To Test

- Try deploying and test normal flow.
- Make some errors on purpose to see if the refactoring of requires to custom errors works properly.
